packages/ckeditor5-modern/src/ModernEditor/ModernEditor.tsx

```diff
@@ -59,10 +66,68 @@ const ModernEditor: FC<ModernEditorProps> = ({
 
   const { isSubmitting } = formState;
   const values = getValues();
+
   const images = useMemo(() => files.map(file => URL.createObjectURL(file)), [
     files,
   ]);
 
+  const uploadFiles = async (files: File[]) => {
+    try {
+      setUploadLoading(true);
+      setUploadCurrentLoading(0);
+      setUploadTotalLoading(files.length * 100);
+      let currentTotal: number[] = [];
+
+      // get upload credentials
+      const resultUploadCredentials: Promise<string>[] = files.map(
+        async (file: File) => {
+          return await uploadGetCrendentials({
+            type: file.type.split('/')[0],
+            filename: file.name,
+            apiUrl: uploadConfig?.apiUrl || '',
+            file: file,
+          });
+        }
+      );
```

Comment:
@nberongoy remove `async` and `await` here, just return the promise right away.

Also you don't have to add type definition for file `file: File` as it is going to infer from the array source.
Also you can omit the typings for `resultUploadCredentials: Promose<string>[]`, let's just infer from the returned array unless otherwise is necessary.